### PR TITLE
Updating SDK references (to fix Dotnetty bug)

### PR DESCRIPTION
**Pull request by vipeller**
Dotnetty had a bug and in some (very rare) cases it got into a tight loop, consuming 100% cpu on a singe core and stopping deallocating memory after some buffering. EdgeHub ran into this bug through the protocol gateway, however it was quicker to update the SDK to reference the fixed version of dotnetty, and through the dependency resolution process, protocol gateway also gets the fixed dotnetty version in edgeHub.
The C# SDK now references dotnetty 0.7.0 (vs 0.6.0).

### CODE REVIEW

File: test/modules/EdgeHubRestartTester/EdgeHubRestartTester.csproj [hunk 1]
```diff
@@ -27,14 +27,14 @@
...
-    <PackageReference Include="Microsoft.Azure.Devices.Client" Version="1.33.1-NestedEdge" />
-    <PackageReference Include="Microsoft.Extensions.Configuration" Version="3.1.3" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="3.1.3" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="3.1.3" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="3.1.3" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="3.1.3" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="3.1.3" />
+    <PackageReference Include="Microsoft.Azure.Devices" Version="1.31.1" />
+    <PackageReference Include="Microsoft.Azure.Devices.Client" Version="1.36.2" />
+    <PackageReference Include="Microsoft.Extensions.Configuration" Version="5.0.0" />
```
Comment (varunpuranik) [inline test/modules/EdgeHubRestartTester/EdgeHubRestartTester.csproj:32]: Do we need to update Config packages to 5.0.0? What are the implications of doing that?

**Comment by vadim-kovalyov** [comment]
Which one is the main PR? this or #5534? It is better to complete one of them and then cherry pick the commit, instead of maintaining 2 PRs. Unless there are a lot of conflicts b/w master and 1.2

**Comment by varunpuranik** [comment]
> Which one is the main PR? this or #5534? It is better to complete one of them and then cherry pick the commit, instead of maintaining 2 PRs. Unless there are a lot of conflicts b/w master and 1.2

+1 - please address comments in the other PR #5534 and once merged, cherry pick that commit instead, to avoid duplicate changes/comments, etc.

**Comment by vipeller** [comment]
with so many reference changes in several project files I felt safer to start changing them from scratch (vs merging from master) I would use this as a baseline and merging the changes to this PR from the one made on the PR for master

**Review by varunpuranik — APPROVED** [review]

**Pull request merged**